File: leetcode/level1/leet1137.js
/**
 * @param {number} n
 * @return {number}
 */
var tribonacci = function(n) {
    let dp = []
    for (let index = 0; index <= n; index++) {
        if (index < 3 ) {
            dp[index] = index == 0 ? 0: 1
            continue
        }
        dp[index] = dp[index-1] + dp[index-2] + dp[index -3] 
    }
    const res = dp.pop()
    // console.log(res);
    return res
};

var testData = [
    // 4,
    25,
]

for (let item of testData) {
    // console.log(item);
    tribonacci(item)
}

File: leetcode/level1/leet1137.py
# py2.7
# n-th-tribonacci-number

class Solution(object):
  def tribonacci(self, n):
    """
    :type n: int
    :rtype: int
    """
    if n == 0:
      return 0
    elif n < 3:
      return 1
    numList = [0,1,1]
    i = 2
    while i < n:
      sum = 0
      for j in numList:
        sum += j
      numList.append(sum)
      numList.pop(0)
      i += 1
    return numList[2]

if __name__ == "__main__":
  testData = [0,1,2,3,4,5]
  test = Solution()
  for item in testData:
    print test.tribonacci(item)
  print test.tribonacci(25)